### ISSUE TITLE
Render saved draft comments for files, inline

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-code-manager#713]
### Describe the problem and steps to reproduce it:

Consider this imaginary UI flow:

- Sign in as a reviewer
- Open a version browse page
- Click on the line number of a file, enter a comment, press save
- Refresh the browser page

### What happened?

n/a

### What did you expect to happen?

The saved draft comments should be displayed

### Anything else we should know?

This depends on https://github.com/mozilla/addons-server/issues/11374

**Comment by EnTeQuAk** [comment]
With mozilla/addons-server#11374 being fixed, I'm going to remove the `needs: api` label for now. Let me know if you need anything specific on top of that.

**Comment by ioanarusiczki** [comment]
@kumar303 I noticed that on manifest.json this works and I could even send with Postman comments, to modify or delete them.
The problem is with the other files, for the add-on I tested with on .txt or .html files after page refresh I did not see the comments posted - so I should investigate it further with another addon and other file types. 

I will get back with more detailed testing results tomorrow.

**Comment by kumar303** [comment]
ok, let me know what the results are. Maybe there were some API errors in the network console? Also, let me know what add-ons you had this trouble with because I'm not totally sure if commenting works on all versions (maybe the version has to be in a specific state).

**Comment by ioanarusiczki** [comment]
@kumar303 

first example: https://code.addons-dev.allizom.org/en-US/browse/617545/versions/1688861/?path=scripts%2Fshare.js. 

I can see after refresh the comments saved for manifest.json but on the rest of the files they won't show up anymore. 

second example: https://code.addons-dev.allizom.org/en-US/browse/491860/versions/1528357/?path=manifest.json 

I can't see the the saved comments anywhere (not even in manifest.json)

third example: https://code.addons-dev.allizom.org/en-US/browse/494431/versions/1532144/?path=devtools.html

For this add-on I see the comments after refreshing the page on all files I tried with. 

I checked on these addons and I see they all have one version in approved state.

**Comment by ioanarusiczki** [comment]
@kumar303
In the Network I did not find errors 
The only errors shown by Consoles are the two below: 

![the only errors](https://user-images.githubusercontent.com/33448286/66652848-597d7600-ec3f-11e9-8f28-6f7f849ca6e1.png)

**Comment by kumar303** [comment]
Do you have exact steps to reproduce the console error *Cannot create key ... because fileName is empty*? I tried adding comments to each of the links up above and I was not able to reproduce this error. Maybe I wasn't commenting on the right file? This error is a bug we need to fix.

**Comment by kumar303** [comment]
Also, keep in mind that if you have left more than 25 comments, no new comments will show up when you refresh the page until https://github.com/mozilla/addons-code-manager/issues/1093 is fixed.

**Comment by kumar303** [comment]
> Do you have exact steps to reproduce the console error _Cannot create key ... because fileName is empty_? 

I found the error in our logs: https://sentry.prod.mozaws.net/operations/addons-code-manager-dev/issues/6449781/?query=is:unresolved%20fileName

It suggests that you were commenting on line 9 of `popup.html` (https://code.addons-dev.allizom.org/en-US/browse/491860/versions/1528357/?path=popup.html#I9) but commenting on that line doesn't reproduce the error for me. Hmm.

**Comment by ioanarusiczki** [comment]
@kumar303 
- That error appears after a page refresh. (I leave a comment than I refreshed the page to see what happens).
I think https://github.com/mozilla/addons-code-manager/issues/1093 is 25 maximum per file so it won't explain why in other files the comments disappear after a page refresh but I'll try with new samples on Monday again and make sure I don't cross the 25 limit per all files and see what happens.

**Comment by kumar303** [comment]
>       I think #1093 is 25 maximum per file so it won't explain why in other files the comments disappear after a page refresh 

The current limitation is to only show 25 comments per version, which includes all files for that version.

**Comment by kumar303** [comment]
>     * That error appears after a page refresh. (I leave a comment than I refreshed the page to see what happens).

Ah, so it is reproducible for you every time? Can you open a new issue for that, including the URL and exact file you commented on?

**Comment by ioanarusiczki** [comment]
@kumar303 
I tested on dev again and here are my results:
- I could make comments on files in different states: waiting approval, approved, disabled, rejected except for a deleted file. 
In case I open a deleted file from the review page I receive a 404 for the draft-comment API on the top of the page, and if I try to make a comment another 404 (same if I try to make a comment on a deleted file from Postman). 

![404](https://user-images.githubusercontent.com/33448286/66754758-5bde0b00-ee9f-11e9-8386-657a1cf77729.gif)

- indeed, if I make more than 25 comments for a version (no matter if it's for a single or multiple files) the other comments will no longer be available after page refresh. 
- I tested with `canned responses` too using Postman - the frontend shows an empty comment in this case but I believe this is to be implemented in the future. 
- should "+" turn into "-" so I could close the text box ? If I must file an issue let me know. Perhaps there will be a "cancel" button later on. 
- for files inside folders when the [API is used](https://addons-server.readthedocs.io/en/latest/topics/api/reviewers.html#drafting-comments) the filename (string|null) needs the path of that file in the Request JSON Object and the documentation is not precise about this. Perhaps that should be updated too. I could file an issue for this.

**Comment by kumar303** [comment]
>     * I could make comments on files in different states: waiting approval, approved, disabled, rejected except for a deleted file.

What do you mean by rejected? Were you able to successfully create comments for each of those states?

>       In case I open a deleted file from the review page I receive a 404 for the draft-comment API on the top of the page, and if I try to make a comment another 404 (same if I try to make a comment on a deleted file from Postman).

Hmm, I'm a bit confused because I am not aware of how to review a deleted file. Could you provide me an example link in the reviewer tool and the exact URL on code manager and the exact file you tried to comment on? In either case, I suggest opening a new bug for this.

>     * should "+" turn into "-" so I could close the text box ? If I must file an issue let me know. Perhaps there will be a "cancel" button later on.

Yeah, we'll add a *cancel* button in https://github.com/mozilla/addons-code-manager/issues/1101

> 
>     * for files inside folders when the [API is used](https://addons-server.readthedocs.io/en/latest/topics/api/reviewers.html#drafting-comments) the filename (string|null) needs the path of that file in the Request JSON Object and the documentation is not precise about this. Perhaps that should be updated too. I could file an issue for this.

Yeah, it's a gotcha. It wouldn't hurt to file an issue in `addons-server` but it might be hard to fix.

**Comment by ioanarusiczki** [comment]
@kumar303 
Rejected = a reviewer is rejecting the file from the add-on review page - which turns to "Disabled" state for the add-on. 

> Were you able to successfully create comments for each of those states?

Yes, I could add comments on different files on that version and after refreshing the page they were still displayed. 

I filed https://github.com/mozilla/addons-code-manager/issues/1130 for deleted files for further investigation. 

Also I filed the documentation issue https://github.com/mozilla/addons-server/issues/12608 hopefully it can be fixed. 

I can mark this verified on dev with FF69(Win10).
